src/sentry/search/events/builder/metrics.py

```diff
@@ -143,15 +157,17 @@ def _on_demand_metric_spec_map(self) -> Optional[Dict[str, OnDemandMetricSpec]]:
         if not self.builder_config.on_demand_metrics_enabled:
             return None
 
-        aggregate_columns = self.selected_columns
         return {
             col: self._get_on_demand_metric_spec(col)
-            for col in aggregate_columns
+            for col in self.selected_columns
```

Comment:
We should only be building specs for on-demand compatible columns, maybe `_get_aggregates` above?

Reply:
I think the `if self._get_on_demand_metric_spec(col)` is what stops it.

Reply:
I tested your idea. It works.